lib/shared/addon/components/cru-cloud-provider/component.js

```diff
@@ -142,21 +170,10 @@ export default Component.extend({
               if (this.isDestroyed || this.isDestroying) {
                 return;
               }
-
               set(this, 'selectedCloudProvider', this.configName);
             });
           }
         }
-      } else {
-        next(() => {
-          if (!this.configName) {
-            if (this.isDestroyed || this.isDestroying) {
-              return;
-            }
-
-            set(this, 'selectedCloudProvider', this.selectedCloudProvider === 'generic' ? 'generic' : 'none');
-          }
-        });
```

Comment:
I couldn't work out what this was attempting to do. It breaks toggling between 'none' and 'external.' I played around with cluster creation/upgrading, with and without templates, couldn't find any problems removing this.

Reply:
This entire function is difficult to understand.  I'll play around some with the template revision so see if I can tease out the intention

Reply:
I never found a way to run into issues with the removal of code either so I figure it's fine until someone else runs into an issue